server/fleet/agent_options_test.go

```diff
@@ -96,6 +96,16 @@ func TestValidateAgentOptions(t *testing.T) {
 				"pack1": 1
 			}
 		}}`, `invalid number value`},
+		{"option added in osquery 5.5.1", `{"config":{
+			"options": {
+				"malloc_trim_threshold": 100
+			}
+		}}`, ``},
```

Comment:
Curious: If the Fleet user sets this and osquery hosts are running < 5.5.1, what happens? Is it ignored?

Reply:
It will return an error, they have to use `fleetctl apply --force` to bypass the validations. This is the intended behaviour.

Reply:
Ah, my question was osquery side, does osquery ignore the config if it doesn't know it?

Reply:
Ah sorry, in this case (using agent options), it prints out a warning and ignore the unknown setting, but that's a good point for the upcoming work on remote flags (where we allow setting command-line flags and we restart osqueryd with the newly set flags) - in that case it flat-out refuses to start if an invalid flag is used. I'll mention this on the ticket for that feature.

Reply:
Nevermind, this was already taken into consideration on #6851 :

> If the new flags are invalid, orbit should restart osquery with the previous valid config set, so as to avoid break in coverage